js/5etools-main.js

```diff
@@ -1331,6 +1332,261 @@ const betteR205etoolsMain = function () {
+						</div>
+					`).appendTo($("body"));
+					const $rdOpt = $dialog.find(`input[type="radio"]`);
+
+					$dialog.dialog({
+						dialogClass: "no-close",
+						buttons: [
+							{
+								text: "Cancel",
+								click: function () {
+									$(this).dialog("close");
+									$dialog.remove();
+									reject(`User cancelled the prompt`);
+								}
+							},
+							{
+								text: "OK",
+								click: function () {
+									const selected = $rdOpt.filter((i, e) => $(e).prop("checked"))
+										.map((i, e) => $(e).data("ix")).get()[0];
+									$(this).dialog("close");
+									$dialog.remove();
+									resolve(selected);
+								}
+							}
+						]
+					})
+				});
+			}
+
+			async function handleProfs(profs, profType) {
+				// Handle the language options, let user choose if needed
+				// Handles most edge cases I think
+				ret = []
+				const profEntries = Object.entries(profs);
+				for (const entry of profEntries) {
+					// Loop must be in this form -- Thanks for figuring this out Giddy
+					const [key, value] = entry;
+					if (key === "choose") {
+						// If choice is needed, call popup function
+						numChoice = 1;
+						if (value.count) numChoice = value.count;
+						const choice = await chooseProfs(value.from, numChoice, profType);
+						choice.forEach(c => ret.push(c));
+					}
+					else if (key === "anyStandard") {
+						// If any language is available, add any
+						for (i = 0; i < value; i++) ret.push("any");
+					}
+					else if (value) {
+						// If no choice is needed, add the proficiency normally
+						ret.push(key);
+					}
+				}
+				return ret;
+			}
+
+			// Get data for language proficiencies specifically
+			let backgroundLanguages = [];
+			if (bg.languageProficiencies && bg.languageProficiencies.length) {
+				if (bg.languageProficiencies.length > 1) {
+					// See Clan Crafter for an example
+					profIndex = await chooseProfsGroup(bg.languageProficiencies, "Languages");
+					backgroundLanguages = await handleProfs(bg.languageProficiencies[profIndex], "Languages");
+				}
+				else if (bg.languageProficiencies.length > 0) {
+					// Most common case
+					backgroundLanguages = await handleProfs(bg.languageProficiencies[0], "Languages");
+				}
+			}
+
+			// Tool Proficiencies
+			let backgroundTools = [];
+			if (bg.toolProficiencies && bg.toolProficiencies.length) {
+				if (bg.toolProficiencies.length > 1) {
+					// If there are different types of options
+					profIndex = await chooseProfsGroup(bg.toolProficiencies, "Tools");
+					backgroundTools = await handleProfs(bg.toolProficiencies[profIndex], "Tools")
+				}
+				else if (bg.toolProficiencies.length > 0) {
+					// Most common case
+					backgroundTools = await handleProfs(bg.toolProficiencies[0] , "Tools");
+				}
+			}
+
+			// Import items
+			async function parseItems(itemlist) {
+				const allitemList = await Renderer.item.pBuildList();
+				const x = Object.values(itemlist).map(function (item) {
```

Comment:
use arrow functions (`.map(item => { ... })`) over `.map(function ...)` to avoid corrupting your `this`